pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -79,6 +79,11 @@ func Add(mgr manager.Manager, ci hcoutil.ClusterInfo) error {
 	return add(mgr, newReconciler(mgr, ci), ci)
 }
 
+// RegisterReconciler creates a new HyperConverged Reconciler and registers it into manager.
+func RegisterReconciler(mgr manager.Manager, ci hcoutil.ClusterInfo) error {
```

Comment:
This is exactly the same as `Add(mgr manager.Manager, ci hcoutil.ClusterInfo) error` - same signature and same implementation. Why do we need both?

Reply:
Beside that, I never understood the original code, so this is simpler and clearer :+1:

Reply:
Mistake from me, `Add` can be removed completely.

We can also think about:
- constructor function vs. direct instantiation e.g. what we have now with `RegisterReconciler` vs. `(&ReconcileHyperConverged{...}).SetupWithManager` (kubebuilder style)
- flatten the package structure in `pkg/controller`